File: python/challenges/distance.py
import math

Point = tuple[int, int]


def point(x: int, y: int) -> Point:
    return Point([x, y])


def distance(a: Point, b: Point) -> float:
    return math.sqrt(((a[0] - b[0]) ** 2) + ((a[1] - b[1]) ** 2))

File: elixir/apps/challenges/lib/challenges/distance.ex
defmodule Challenges.Distance do
  @type point() :: {integer(), integer()}

  @spec distance(point(), point()) :: float()
  def distance(a, b) do
    {{ax, ay}, {bx, by}} = {a, b}
    :math.sqrt(((ax - bx)**2) + ((ay - by)**2))
  end
end
